### PR TITLE
Update to SDK v0.40.0

### PR DESCRIPTION
**Pull request by amaurymartiny**

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -287,7 +288,7 @@ func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest
 
 	// Create IBC Keeper
 	app.IBCKeeper = ibckeeper.NewKeeper(
-		appCodec, keys[ibchost.StoreKey], app.StakingKeeper, scopedIBCKeeper,
+		appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, scopedIBCKeeper,
```
Comment (amaurymartiny) [inline app/app.go:291]: is IBC needed actually for regen-ledger for mainnet?

Comment (anilCSE) [reply app/app.go:290]: I don't see a reason for not having that. We can enable IBC transfers via governance when we feel comfortable.

File: app/app_test.go [hunk 2]
```diff
@@ -6,45 +6,40 @@ import (
 	"testing"
```
Comment (amaurymartiny) [inline app/app_test.go:7]: I see a `WARNING: DATA RACE` in the tests. Does anyone have an idea how to fix this?

File: app/app.go [hunk 3]
```diff
@@ -206,9 +207,9 @@ type RegenApp struct {
 func NewRegenApp(logger log.Logger, db dbm.DB, traceStore io.Writer, loadLatest bool, skipUpgradeHeights map[int64]bool,
 	homePath string, invCheckPeriod uint, encodingConfig simappparams.EncodingConfig, baseAppOptions ...func(*baseapp.BaseApp)) *RegenApp {
 
-	// TODO: Remove cdc in favor of appCodec once all modules are migrated.
+	// TODO: Remove cdc legacyAmino favor of appCodec once all modules are migrated.
```
Comment (robert-zaremba) [inline app/app.go:210]:
```suggestion
	// TODO: Remove cdc legacyAmino in favor of appCodec once all modules are migrated.
```

**Review by robert-zaremba — APPROVED** [review]

File: app/app_test.go (hunk 2 above)
Comment (amaurymartiny) [reply app/app_test.go:7]: Oh well, Robert's grammar change (https://github.com/regen-network/regen-ledger/pull/219#discussion_r557418453) made the error go away 👍

**Review by anilCSE — APPROVED** [review]
lgtm

**Pull request merged**